railties/test/generators/app_generator_test.rb

```diff
@@ -383,6 +383,25 @@ def test_app_update_does_not_change_config_target_version
     assert_file "config/application.rb", /\s+config\.load_defaults 5\.1/
   end
 
+  def test_app_update_does_not_change_app_name_when_app_name_is_hyphenated_name
+    app_root = File.join(destination_root, "hyphenated-app")
+    run_generator [app_root]
+
+    assert_file "#{app_root}/config/cable.yml" do |content|
+      assert_match(/hyphenated-app/, content)
+      assert_no_match(/hyphenated_app/, content)
+    end
+
+    FileUtils.cd(app_root) do
+      quietly { system("bin/rails app:update") }
```

Comment:
The name of app_root is not always matching with app_name (renamed after created, git checkout working dir specified, etc.), if so, will it works as expected?

Reply:
Your point is right. But do you think it's better to disallow `-`, when we run `rails new`? If so (and we want to unify these difference), we have to give priority to const name over initial args.

https://github.com/rails/rails/blob/master/railties/lib/rails/generators/rails/app/app_generator.rb#L492-L494

I thought it's better to allow hyphen (because many apps given hyphened name might exists, I suppose.)

Reply:
It might be better to replace hyphen with underscore, I thought. So I'm going to modify it.

Reply:
Changed be prefer to const name than directory name.